baybe/targets/binary.py

```diff
@@ -0,0 +1,91 @@
+"""Binary targets."""
+
+from typing import TypeAlias
+
+import numpy as np
+import pandas as pd
+from attrs import define, field
+from attrs.validators import instance_of
+
+from baybe.exceptions import InvalidTargetValueError
+from baybe.serialization import SerialMixin
+from baybe.targets.base import Target
+
+ChoiceValue: TypeAlias = bool | int | float | str
+"""Types of values that a :class:`BinaryTarget` can take."""
+
+_POSITIVE_VALUE_COMP = True
+"""Computational representation of the positive choice value."""
+
+_NEGATIVE_VALUE_COMP = False
+"""Computational representation of the negative choice value."""
+
+
+@define(frozen=True)
+class BinaryTarget(Target, SerialMixin):
+    """Class for binary targets."""
+
+    # FIXME[typing]: https://github.com/python-attrs/attrs/issues/1336
+
+    positive_value: ChoiceValue = field(
+        default=1,
+        validator=instance_of(ChoiceValue),  # type: ignore[call-overload]
+        kw_only=True,
+    )
+    """Experimental representation of the positive value."""
+
+    negative_value: ChoiceValue = field(
+        default=0,
+        validator=instance_of(ChoiceValue),  # type: ignore[call-overload]
+        kw_only=True,
+    )
+    """Experimental representation of the negative value."""
+
+    @negative_value.validator
+    def _validate_values(self, _, value):
+        """Validate that the two choice values of the target are different."""
+        if value == self.positive_value:
+            raise ValueError(
+                f"The two choice values of a '{BinaryTarget.__name__}' must be "
+                f"different but the following value was provided for both choices of "
+                f"target '{self.name}': {value}"
+            )
+
+    def transform(self, data: pd.DataFrame) -> pd.DataFrame:  # noqa: D102
+        # TODO: The method (signature) needs to be refactored, potentially when
```

Comment:
Yes, fully agree 👍🏼 But if you can live with it for one more month, I'd prefer to do this properly as a preparation PR before starting the actual multi-target PR. I'd like to clean a few things up and make it consistent to how we handle things in the parameter/searchspace classes. That is, for example:
* Making this method expect a series instead of df
* Renaming stuff to our current comment terminology / making a few things positional-only
* Adding explicit handling how to treat additional unncessary columns etc.

So basically all the stuff we cleaned up for the parameters. And that is out of the scope of this PR. Accept?

Reply:
ok, shifted to multi-target